features/element-hiding/main.js

```diff
@@ -0,0 +1,128 @@
+const TEST_DOMAIN = 'good.third-party.site';
+const expectedResults = [
+    { id: 'hide-basic', description: 'hide rule: element containing text is hidden', hidden: true },
+    { id: 'hide-frame-contents', description: 'hide rule: element containing iframe is hidden', hidden: true },
+    { id: 'hide-delayed-frame', description: 'hide rule: element that loads 3p frame after delay is hidden', hidden: true },
+    { id: 'hide-delayed', description: 'hide rule: element added to page after delay is hidden', hidden: true },
+    { id: 'hide-empty-no-content', description: 'hide-empty rule: element containing no content is hidden', hidden: true },
+    { id: 'hide-empty-ad-label', description: 'hide-empty rule: element containing ad label content is hidden', hidden: true },
+    { id: 'hide-empty-text-content', description: 'hide-empty rule: element containing text content is not hidden', hidden: false },
+    { id: 'hide-empty-frame-content', description: 'hide-empty rule: element containing iframe is not hidden', hidden: false },
+    { id: 'hide-empty-delayed-frame', description: 'hide-empty rule: element that loads 3p frame after delay is not hidden', hidden: false },
+    { id: 'hide-empty-delayed', description: 'hide-empty rule: empty element added to page after delay is hidden', hidden: true },
+    { id: 'closest-empty-single-div', description: 'closest-empty rule: element containing no content is hidden', hidden: true },
+    { id: 'closest-empty-single-nested-div', description: 'closest-empty rule: parent of target element is hidden', hidden: true },
+    { id: 'closest-empty-siblings-nested-div', description: 'closest-empty rule: parent of target element with empty siblings is hidden', hidden: true },
+    { id: 'closest-empty-siblings-content-nested-div', description: 'closest-empty rule: parent of target element with non-empty siblings is not hidden', hidden: false },
+    { id: 'closest-empty-frame-content', description: 'closest-empty rule: parent of element containing iframe is not hidden', hidden: false },
+    { id: 'closest-empty-delayed-frame', description: 'closest-empty rule: parent of element that loads 3p frame after delay is not hidden', hidden: false },
+    { id: 'closest-empty-intermediate-element', description: 'closest-empty rule: only outermost element should be hidden when target is nested', hidden: false },
+    { id: 'closest-empty-delayed-parent', description: 'closest-empty rule: parent of empty element added to page after delay is hidden', hidden: true },
```

Comment:
not sure if that's the test, but are we testing the "unhiding" after content is loaded into a frame that we thought is empty?

Reply:
So this one is actually testing that the parent element (in this case, the most distant ancestor that has no visible content) is hidden after an element we target with a `closest-empty-test` class is added to it.

Let me add a test for the case when a frame we thought was empty has content added to it.